core/src/main/java/org/apache/calcite/rel/logical/LogicalCalc.java

```diff
@@ -124,4 +138,9 @@ public static LogicalCalc create(final RelNode input,
     }
     variableSet.addAll(vuv.variables);
   }
+
+  @Override public RelNode withHints(List<RelHint> hintList) {
+    return new LogicalCalc(getCluster(), traitSet,
+        ImmutableList.copyOf(hintList), input, program);
+  }
```

Comment:
We do not need invoke `ImmutableList.copyOf(` here, the constructor of `LogicalCalc` already did that. But in order of code safety, i think we can keep it here.